### ISSUE TITLE
xrdp hang issue

### ISSUE THREAD

**Issue by purvalpatel** [neutrinolabs/xrdp#925]
hello,
i have ubuntu server. and 40 users are connect server via xrdp. i am using xrdp 0.9.0 version.
two times in a day xrdp process is hang.
users just connect with the server with RDP. enter the user name and password and when they are going to click on ok button. the button even not getting clicked.it was hanged.

the users who is already connected is remain connected and using it smoothly, but if they logout and try to login again, then they are not able to login.

**Comment by moobyfr** [comment]
which backend are you using ?
what's in xrdp.log and xrdp-sesman.log ?

**Comment by purvalpatel** [comment]
xrdp-sesman.log

[20171026-14:50:21] [INFO ] A connection received from: 127.0.0.1 port 57221
[20171026-14:50:21] [INFO ] scp thread on sck 8 started successfully
[20171026-14:50:21] [INFO ] ++ created session (access granted): username ttplrkl, ip 117.205.183.68:3533 - socket: 8
[20171026-14:50:21] [INFO ] starting X11rdp session...
[20171026-14:50:21] [INFO ] An established connection closed to endpoint: NULL:NULL - socket: 9
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1


xrdp.log

[20171026-14:50:21] [INFO ] An established connection closed to endpoint: 109.112.47.46:12148 - socket: 13
[20171026-14:50:21] [INFO ] An established connection closed to endpoint: 117.200.235.201:11412 - socket: 8
[20171026-14:50:21] [DEBUG] xrdp_mm_module_cleanup
[20171026-14:50:21] [INFO ] An established connection closed to endpoint: 109.112.47.46:12148 - socket: 12
[20171026-14:50:21] [ERROR] Listening socket is in wrong state we terminate listener
[20171026-14:50:24] [INFO ] A connection received from: 117.200.235.201 port 11931
[20171026-14:50:24] [INFO ] An established connection closed to endpoint: 117.200.235.201:11931 - socket: 8
[20171026-14:50:24] [INFO ] An established connection closed to endpoint: NULL:NULL - socket: 7
[20171026-14:50:24] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/cert.pem
[20171026-14:50:24] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/key.pem
[20171026-14:50:25] [INFO ] An established connection closed to endpoint: 117.200.235.201:11931 - socket: 8
[20171026-14:50:25] [ERROR] Listening socket is in wrong state we terminate listener
[20171026-14:50:25] [INFO ] A connection received from: 117.200.235.201 port 11932

using ubuntu 14.04 32bit

**Comment by purvalpatel** [comment]
xrdp.log

[20171026-14:50:21] [INFO ] An established connection closed to endpoint: 109.112.47.46:12148 - socket: 13
[20171026-14:50:21] [INFO ] An established connection closed to endpoint: 117.200.235.201:11412 - socket: 8
[20171026-14:50:21] [DEBUG] xrdp_mm_module_cleanup
[20171026-14:50:21] [INFO ] An established connection closed to endpoint: 109.112.47.46:12148 - socket: 12
[20171026-14:50:21] [ERROR] Listening socket is in wrong state we terminate listener
[20171026-14:50:24] [INFO ] A connection received from: 117.200.235.201 port 11931
[20171026-14:50:24] [INFO ] An established connection closed to endpoint: 117.200.235.201:11931 - socket: 8
[20171026-14:50:24] [INFO ] An established connection closed to endpoint: NULL:NULL - socket: 7
[20171026-14:50:24] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/cert.pem
[20171026-14:50:24] [CORE ] WARNING: Invalid x.509 certificate path defined, default path will be used: /etc/xrdp/key.pem
[20171026-14:50:25] [INFO ] An established connection closed to endpoint: 117.200.235.201:11931 - socket: 8
[20171026-14:50:25] [ERROR] Listening socket is in wrong state we terminate listener
[20171026-14:50:25] [INFO ] A connection received from: 117.200.235.201 port 11932


xrdp-sesman.log

[20171026-14:50:21] [INFO ] A connection received from: 127.0.0.1 port 57221
[20171026-14:50:21] [INFO ] scp thread on sck 8 started successfully
[20171026-14:50:21] [INFO ] ++ created session (access granted): username ttplrkl, ip 117.205.183.68:3533 - socket: 8
[20171026-14:50:21] [INFO ] starting X11rdp session...
[20171026-14:50:21] [INFO ] An established connection closed to endpoint: NULL:NULL - socket: 9
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1
[20171026-15:29:06] [INFO ] shutting down sesman 1

using xrdp in ubuntu 14.04 32 bit system

**Comment by purvalpatel** [comment]
40 users are login daily on ubuntu server, and after some times xrdp login screen is stuck. no user can login.

here my /etc/xrdp/xrdp.ini

```
[globals]
# xrdp.ini file version number
ini_version=1

bitmap_cache=yes
bitmap_compression=yes
port=3389
allow_channels=true
channel_code=1
max_bpp=8
fork=yes
# minimum security level allowed for client
# can be 'none', 'low', 'medium', 'high', 'fips'
crypt_level=high
# security layer can be 'tls', 'rdp' or 'negotiate'
# for client compatible layer
security_layer=rdp
# X.509 certificate and private key
# openssl req -x509 -newkey rsa:2048 -nodes -keyout key.pem -out cert.pem -days 365
certificate=
key_file=

# regulate if the listening socket use socket option tcp_nodelay
# no buffering will be performed in the TCP stack
tcp_nodelay=yes

# regulate if the listening socket use socket option keepalive
# if the network connection disappear without close messages the connection will be closed
tcp_keepalive=yes

#tcp_send_buffer_bytes=32768
#tcp_recv_buffer_bytes=32768

#
# colors used by windows in RGB format
#

blue=009cb5
grey=dedede
#black=000000
#dark_grey=808080
#blue=08246b
#dark_blue=08246b
#white=ffffff
#red=ff0000
#green=00ff00
#background=626c72
#autorun=xrdp1
#hidelogwindow=yes

# when true, userid/password *must* be passed on cmd line
# require_credentials=yes

# set a default entry for autorun if the client send login and pass directly
autorun=xrdp1

bulk_compression=yes

# You can set the PAM error text in a gateway setup (MAX 256 chars)
#pamerrortxt=change your password according to policy at http://url
new_cursors=yes
allow_multimon=true

# fastpath - can be set to input / output / both / none
use_fastpath=both
#
# configure login screen
#

# Login Screen Window Title
ls_title=Login Tally Cloud

# top level window background color in RGB format
ls_top_window_bg_color=009cb5

# width and height of login screen
ls_width=350
ls_height=430

# login screen background color in RGB format
ls_bg_color=dedede

# optional background image filename (bmp format).
#ls_background_image=

# logo
# full path to bmp-file or file in shared folder
ls_logo_filename=
ls_logo_x_pos=55
ls_logo_y_pos=50

# for positioning labels such as username, password etc
ls_label_x_pos=30
ls_label_width=60

# for positioning text and combo boxes next to above labels
ls_input_x_pos=110
ls_input_width=210

# y pos for first label and combo box
ls_input_y_pos=220

# OK button
ls_btn_ok_x_pos=142
ls_btn_ok_y_pos=370
ls_btn_ok_width=85
ls_btn_ok_height=30

# Cancel button
ls_btn_cancel_x_pos=237
ls_btn_cancel_y_pos=370
ls_btn_cancel_width=85
ls_btn_cancel_height=30

[Logging]
LogFile=xrdp.log
LogLevel=DEBUG
EnableSyslog=1
SyslogLevel=DEBUG
# LogLevel and SysLogLevel could by any of: core, error, warning, info or debug

[channels]
# Channel names not listed here will be blocked by XRDP.
# You can block any channel by setting its value to false.
# IMPORTANT! All channels are not supported in all use
# cases even if you set all values to true.
# You can override these settings on each session type
# These settings are only used if allow_channels=true
rdpdr=true
rdpsnd=true
drdynvc=true
cliprdr=true
rail=true
xrdpvr=true
tcutils=true

# for debugging xrdp, in section xrdp1, change port=-1 to this:
# port=/tmp/.xrdp/xrdp_display_10

# for debugging xrdp, add following line to section xrdp1
# chansrvport=/tmp/.xrdp/xrdp_chansrv_socket_7210

[xrdp1]
name=sesman-X11rdp
lib=libxup.so
username=ask
password=ask
ip=127.0.0.1
port=-1
xserverbpp=24
code=10

[xrdp2]
name=sesman-Xvnc
lib=libvnc.so
username=ask
password=ask
ip=127.0.0.1
port=-1
#delay_ms=2000

[xrdp3]
name=console
lib=libvnc.so
ip=127.0.0.1
port=5900
username=na
password=ask
#delay_ms=2000

[xrdp4]
name=vnc-any
lib=libvnc.so
ip=ask
port=ask5900
username=na
password=ask
#pamusername=asksame
#pampassword=asksame
#pamsessionmng=127.0.0.1
#delay_ms=2000

[xrdp5]
name=sesman-any
lib=libvnc.so
ip=ask
port=-1
username=ask
password=ask
#delay_ms=2000

[xrdp6]
name=rdp-any
lib=librdp.so
ip=ask
port=ask3389

[xrdp7]
name=neutrinordp-any
lib=libxrdpneutrinordp.so
ip=ask
port=ask3389
username=ask
password=ask

[Session manager]
name=Session manager
lib=libxup.so
username=ask
password=ask
ip=127.0.0.1
port=-1
xserverbpp=24
code=20

# You can override the common channel settings for each session type
#channel.rdpdr=true
#channel.rdpsnd=true
#channel.drdynvc=true
#channel.cliprdr=true
#channel.rail=true
#channel.xrdpvr=true

```

sesman.ini
```
[Globals]
ListenAddress=127.0.0.1
ListenPort=3350
EnableUserWindowManager=1
UserWindowManager=startwm.sh
DefaultWindowManager=startwm.sh

[Security]
AllowRootLogin=1
MaxLoginRetry=4
TerminalServerUsers=tsusers
TerminalServerAdmins=tsadmins
# When AlwaysGroupCheck = false access will be permitted
# if the group TerminalServerUsers is not defined.
AlwaysGroupCheck = false

[Sessions]

## X11DisplayOffset - x11 display number offset
# Type: integer
# Default: 10
X11DisplayOffset=10

## MaxSessions - maximum number of connections to an xrdp server
# Type: integer
# Default: 0
MaxSessions=200

## KillDisconnected - kill disconnected sessions
# Type: integer
# Default: 0
# if 1, true, or yes, kill session after 60 seconds
KillDisconnected=0

## IdleTimeLimit - when to disconnect idle sessions
# Type: integer
# Default: 0
# if not zero, the seconds without mouse or keyboard input before disconnect
# not complete yet
IdleTimeLimit=0

## DisconnectedTimeLimit - when to kill idle sessions
# Type: integer
# Default: 0
# if not zero, the seconds before a disconnected session is killed
# min 60 seconds
DisconnectedTimeLimit=0

## Policy - session allocation policy
# Type: enum [ "Default" | "UBD" | "UBI" | "UBC" | "UBDI" | "UBDC" ]
# Default: Xrdp:<User,BitPerPixel> and Xvnc:<User,BitPerPixel,DisplaySize>
# "UBD" session per <User,BitPerPixel,DisplaySize>
# "UBI" session per <User,BitPerPixel,IPAddr>
# "UBC" session per <User,BitPerPixel,Connection>
# "UBDI" session per <User,BitPerPixel,DisplaySize,IPAddr>
# "UBDC" session per <User,BitPerPixel,DisplaySize,Connection>
Policy=Default

[Logging]
LogFile=xrdp-sesman.log
LogLevel=DEBUG
EnableSyslog=1
SyslogLevel=DEBUG

[X11rdp]
param1=-bs
param2=-ac
param3=-nolisten
param4=tcp
param5=-uds
param6=-once

[Xvnc]
param1=-bs
param2=-ac
param3=-nolisten
param4=tcp
param5=-localhost
param6=-dpi
param7=96

[Xorg]
param1=-config
param2=xrdp/xorg.conf
param3=-logfile
param4=/dev/null
param5=-noreset
param6=-ac
param7=-nolisten
param8=tcp

[Chansrv]
# drive redirection, defaults to xrdp_client if not set
FuseMountName=thinclient_drives

[SessionVariables]
PULSE_SCRIPT=/etc/xrdp/pulse/default.pa

```

**Comment by johnarnold** [comment]
Try a newer version, lots of bug fixes.

**Comment by metalefty** [comment]
No feedback from the reporter for 1 month, closing. If the issue still occurs with the latest version, please come back. I'll reopen this.

**Comment by gohalien** [comment]
I have this issue running xrdp in raspbian, sometimes it freeze after successful/failedl login (click ok button)
xrdp.log

`[20171214-03:23:10] [INFO ] A connection received from: ::ffff:192.168.0.200 port 52709
[20171214-03:23:10] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.0.187 port 3389)
[20171214-03:23:10] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20171214-03:23:10] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20171214-03:23:10] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20171214-03:23:10] [DEBUG] Security layer: requested 11, selected 0
[20171214-03:23:10] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.0.187 port 3389)
[20171214-03:23:10] [ERROR] Listening socket is in wrong state, terminating listener
[20171214-03:23:10] [INFO ] A connection received from: ::ffff:192.168.0.200 port 52710
[20171214-03:23:10] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.0.187 port 3389)
[20171214-03:23:10] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20171214-03:23:10] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20171214-03:23:10] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20171214-03:23:10] [DEBUG] Security layer: requested 0, selected 0
[20171214-03:23:11] [DEBUG] xrdp_000011db_wm_login_mode_event_00000001
[20171214-03:23:11] [INFO ] Loading keymap file /etc/xrdp/km-0000040a.ini
[20171214-03:23:11] [WARN ] local keymap file for 0x0000040a found and doesn't match built in keymap, using local keymap file
[20171214-03:23:22] [DEBUG] xrdp_wm_log_msg: connecting to sesman ip 127.0.0.1 port 3350
[20171214-03:23:22] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:22] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:23] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:23] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:24] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:25] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:26] [DEBUG] Closed socket 16 (AF_INET6 :: port 0)
[20171214-03:23:26] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:26] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:27] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:27] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:28] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:29] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:30] [DEBUG] Closed socket 16 (AF_INET6 :: port 0)
[20171214-03:23:30] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:30] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:31] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:31] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:32] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:33] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:34] [DEBUG] Closed socket 16 (AF_INET6 :: port 0)
[20171214-03:23:34] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:34] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:35] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:36] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:36] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:37] [ERROR] g_tcp_connect: getaddrinfo() failed: Name or service not known
[20171214-03:23:38] [ERROR] xrdp_wm_log_msg: Error connecting to sesman: 127.0.0.1 port: 3350
[20171214-03:23:38] [DEBUG] Closed socket 16 (AF_INET6 :: port 0)
[20171214-03:23:38] [DEBUG] return value from xrdp_mm_connect 1
[20171214-03:23:38] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.0.187 port 3389)
[20171214-03:23:38] [DEBUG] xrdp_mm_module_cleanup
[20171214-03:23:38] [ERROR] Listening socket is in wrong state, terminating listener`

xrdp-sesman.log (There is no log when I tried to log in at 3:23, I rebooted rasperry pi and logged in successfully at 3:30)
`[20171214-03:17:07] [DEBUG] libscp initialized
[20171214-03:17:07] [INFO ] starting xrdp-sesman with pid 416
[20171214-03:17:07] [INFO ] listening to port 3350 on 127.0.0.1
[20171214-03:30:12] [INFO ] A connection received from ::1 port 46540
[20171214-03:30:14] [DEBUG] Closed socket 8 (AF_INET6 ::1 port 3350)
[20171214-03:30:25] [INFO ] A connection received from ::1 port 46542
[20171214-03:30:26] [INFO ] ++ created session (access granted): username pi, ip 0.0.0.0:52963 - socket: 12
[20171214-03:30:26] [INFO ] starting Xorg session...
[20171214-03:30:26] [DEBUG] Closed socket 9 (AF_INET6 :: port 5910)
[20171214-03:30:26] [DEBUG] Closed socket 9 (AF_INET6 :: port 6010)
[20171214-03:30:26] [DEBUG] Closed socket 9 (AF_INET6 :: port 6210)
[20171214-03:30:26] [DEBUG] Closed socket 8 (AF_INET6 ::1 port 3350)
[20171214-03:30:26] [DEBUG] Closed socket 7 (AF_INET6 ::1 port 3350)
[20171214-03:30:26] [INFO ] Xorg :10 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp  
[20171214-03:30:26] [INFO ] starting xrdp-sessvc - xpid=767 - wmpid=766`
